docs/getting_started.rst

```diff
@@ -0,0 +1,283 @@
+* :code:`dyad::wofstream_dyad`
+* :code:`dyad::basic_fstream_dyad`
+* :code:`dyad::fstream_dyad`
+* :code:`dyad::wfstream_dyad`
+
+When using DYAD, these file streams should be used in place of the file streams from the C++
+Standard Library. The DYAD file streams should be directly used to do the following:
+
+* Open files (with the file stream's :code:`open` method)
+* Close files (with the file stream's :code:`close` method or destructor)
+* Access the underlying C++ Standard Library file stream using the DYAD stream's :code:`get_stream` method
+
+All reading from and writing to files should be done using the underlying C++ Standard Library file stream.
+A simple example of using DYAD's C++ API in a producer application is shown below:
+
+.. code-block:: cpp
+
+   #include <dyad_stream_api.hpp>
+
+   void produce_file(std::string& full_path, int32_t* data, std::size_t data_size)
+   {
+       dyad::ofstream_dyad ofs_dyad;
+       ofs_dyad.open(full_path, std::ofstream::out | std::ios::binary);
+       std::ofstream& ofs = ofs_dyad.get_stream();
+       ofs.write((char*) data, data_size);
+       ofs_dyad.close();
+   }
+
+After replacing C++ Standard Library file streams with their DYAD equivalents,
+there is one final requirement to using the C++ API. When compiling your code,
+you must link the associated library (i.e., :code:`libdyad_stream.so` or
+:code:`libdyad_stream.a`). This library can be found in the :code:`lib`
+subdirectory of the install prefix.
+
+Running DYAD
+############
+
+There are three steps to running DYAD-enabled applications:
+
+1. :ref:`Create a Flux key-value store (KVS) namespace <Create a Flux KVS Namespace>`
+2. :ref:`Determine the managed directories for each application <Determine the Managed Directories for Each Application>`
+3. :ref:`Load DYAD's Flux module <Load DYAD's Flux Module>`
+4. :ref:`Configure and run the DYAD-enabled applications <Configure and Run the DYAD-Enabled Applications>`
+
+Create a Flux KVS Namespace
+***************************
+
+The first step in running DYAD is to create a namespace in the Flux's hierarchical key-value store (KVS).
+This namespace is used by DYAD to exchange file information (e.g., the Flux broker that "owns"
+a file) needed to synchronize the consumer application and transfer the file from producer to
+consumer. To create this namespace, run the following:
+
+.. code-block:: shell
+
+   $ flux kvs namespace create <DYAD_KVS_NAMESPACE>
+
+The namespace can be whatever string value you want.
+
+Determine the Managed Directories for Each Application
+******************************************************
+
+To determine when to perform synchronization and data transfer, DYAD tracks two directories for
+each application: a **producer-managed directory** and a **consumer-managed directory**. At least
+one of these directories must be specified for DYAD to do anything. If neither are provided, the application
+will still run, but DYAD will not do anything.
```

Comment:
Mention that DYAD will not interfere the I/O operation on files under a path that is not managed.

Reply:
This is already added as a note at the end of this section. I chose to do this as a note to make it stand out to anyone reading the page.

Reply:
I see. All good then.